From 1c4bdc31db36301a5ec195c54b76ecf3c3068d11 Mon Sep 17 00:00:00 2001
From: "Simental Magana, Marcos" <marcos.simental.magana@intel.com>
Date: Wed, 7 Oct 2015 13:29:40 -0500
Subject: [PATCH] Fix OrderedDict mutated during iteration bug

Please check
https://bugs.launchpad.net/python-keystoneclient/+bug/1483872
as this is an upstream bug which is being fixed atm, so probably
would be fixed in next release.
---
 keystoneclient/session.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/keystoneclient/session.py b/keystoneclient/session.py
index e3db3b9..ae640b6 100644
--- a/keystoneclient/session.py
+++ b/keystoneclient/session.py
@@ -135,7 +135,7 @@ class Session(object):
         if not session:
             session = requests.Session()
             # Use TCPKeepAliveAdapter to fix bug 1323862
-            for scheme in session.adapters:
+            for scheme in list(session.adapters):
                 session.mount(scheme, TCPKeepAliveAdapter())
 
         self.auth = auth
-- 
2.4.3

